src/hooks/useQuiz.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import ConstructorIOClient from '@constructor-io/constructorio-client-javascript';
+import { GetBrowseResultsResponseData } from '@constructor-io/constructorio-client-javascript/lib/types';
+import { QuizAPIReducerState } from '../components/CioQuiz/quizApiReducer';
+import { QuizLocalReducerState } from '../components/CioQuiz/quizLocalReducer';
+import { ResultsPageOptions } from '../components/Results/Results';
+import useCioClient from './useCioClient';
+import useConsoleErrors from './useConsoleErrors';
+import useQuizApiState from './useQuizApiState';
+import useQuizLocalState from './useQuizLocalState';
+import useQuizTrackingAndCbEvents from './useQuizTrackingAndCbEvents';
+
+export interface IQuizProps {
+  apiKey?: string;
+  cioJsClient?: ConstructorIOClient;
+  quizId: string;
+  quizVersionId?: string;
+  resultsPageOptions: ResultsPageOptions;
+}
+
+export interface UseQuizReturnEvents {
+  quizNextHandler: () => void;
+  quizBackHandler: () => void;
+  resetQuizClickHandler: () => void;
+  resultClickHandler: (result: Partial<GetBrowseResultsResponseData>, position: number) => void;
+  addToCartClickHandler: (
+    e: React.MouseEvent<HTMLElement>,
+    result: Partial<GetBrowseResultsResponseData>,
+    price: any
+  ) => void;
+}
+
+export interface UseQuizReturn {
+  cioClient?: ConstructorIOClient;
+  quizLocalState?: QuizLocalReducerState;
+  quizApiState?: QuizAPIReducerState;
+  isFirstQuestion?: boolean;
+  events: UseQuizReturnEvents;
+}
+
+type UseQuiz = (quizProps: IQuizProps) => UseQuizReturn;
+
+const useQuiz: UseQuiz = ({ quizId, apiKey, cioJsClient, quizVersionId, resultsPageOptions }) => {
+  // Log console errors for required parameters quizId and resultsPageOptions
+  useConsoleErrors(quizId, resultsPageOptions);
```

Comment:
Why does this need to be a hook? Can't we just use a regular function?

Reply:
I thought this at first as well but following react best practices by having it within a useEffect that only run when any of it's dependency arrays get updated. If we use a regular function it will be called on every render. We can have the logic in a util function like logConsoleErrors() but we still will call it into a useEffect. If you prefer this better I'm open to change it I don't have a preference just that it has to be called in a useEffect